spec/rollbar_helper_spec.rb

```diff
@@ -7,43 +7,68 @@
+  end
+
+
   it "has a version number" do
     expect(RollbarHelper::VERSION).not_to be nil
   end
 
   it 'responds to log level methods' do
+    expect(RollbarHelper.respond_to?(:debug)).to    eq(true)
+    expect(RollbarHelper.respond_to?(:info)).to     eq(true)
+    expect(RollbarHelper.respond_to?(:warning)).to  eq(true)
+    expect(RollbarHelper.respond_to?(:error)).to    eq(true)
     expect(RollbarHelper.respond_to?(:critical)).to eq(true)
-    expect(RollbarHelper.respond_to?(:debug)).to eq(true)
-    expect(RollbarHelper.respond_to?(:error)).to eq(true)
-    expect(RollbarHelper.respond_to?(:info)).to eq(true)
-    expect(RollbarHelper.respond_to?(:warning)).to eq(true)
+  end
+
+  it 'support sending both a message and an exception' do
+    expect(Rollbar).to receive(:error).with('Oops!', instance_of(StandardError), data).and_call_original
+    expect_any_instance_of(Rollbar::Item).to receive(:build_backtrace_body).and_call_original
+    RollbarHelper.error('Oops!', StandardError.new('Error'), data)
+  end
+
+  context '#core legacy' do
+    it 'support sending both a message and an exception' do
+      expect(Rollbar).to receive(:error).with('Oops!', instance_of(StandardError), data).and_call_original
+      expect_any_instance_of(Rollbar::Item).to receive(:build_backtrace_body).and_call_original
+      RollbarHelper.error('Oops!', data.merge(e: StandardError.new('Error')))
+    end
   end
 
   it 'wraps rollbar method with error' do
-    expect(Rollbar).to receive(:error).with(instance_of(StandardError), :data => true)
-    RollbarHelper.error('Oops!', :data => true)
+    expect(Rollbar).to receive(:error).with(nil, instance_of(StandardError), data).and_call_original
+    expect_any_instance_of(Rollbar::Item).to receive(:build_backtrace_body).and_call_original
+    RollbarHelper.error('Oops!', data)
   end
 
   it 'uses callers backtrace' do
-    expect(Rollbar).to receive(:error).with(caller_backtrace(__FILE__), :data => true)
-    RollbarHelper.error('Oops!', :data => true)
+    expect(Rollbar).to receive(:error).with(nil, caller_backtrace(__FILE__), data).and_call_original
+    expect_any_instance_of(Rollbar::Item).to receive(:build_backtrace_body).and_call_original
+    RollbarHelper.error('Oops!', data)
   end
 
   it 'scopes with fingerprint' do
-    expect(Rollbar).to receive(:scope).with(:fingerprint => 'processor').and_return(Rollbar)
-    expect(Rollbar).to receive(:error).with(caller_backtrace(__FILE__), :data => true)
-    RollbarHelper.error('Oops!', :fingerprint => 'processor', :data => true)
+    expect(Rollbar).to receive(:scope).with(:fingerprint => 'processor').and_call_original
+    expect_any_instance_of(Rollbar::Notifier).to receive(:error).with(nil, caller_backtrace(__FILE__), data).and_call_original
+    expect_any_instance_of(Rollbar::Item).to receive(:build_backtrace_body).and_call_original
+    RollbarHelper.error('Oops!', data, :fingerprint => 'processor')
```

Comment:
Looks like the repo was using hash rocket, but now we're changing it to the symbol notation except in specs that is still using hash rocket? nbd but preferably we'd just stick to one within the repo

Reply:
Updating to hash syntax